.github/workflows/test-nix.yml

```
@@ -0,0 +1,36 @@
+---
+name: Test Nix package and environment
+
+on:
+  schedule:
+    - cron: '0 1 * * 1'
+  push:
+    tags:
+      - '*'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
```

A little nit, as I won't have to do another PR later on.
There is a pattern to explicitly clear all permissions for the workflow, then per-job to select which ones to add (instead of having read all as default)
```suggestion

permissions: {}

```

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs